src/core/lib/surface/call.cc

```diff
@@ -2244,13 +2297,24 @@ class PromiseBasedCall : public Call,
   grpc_call_stats final_stats_{};
   CallFinalization finalization_;
   // Current deadline.
-  Timestamp deadline_ = Timestamp::InfFuture();
-  grpc_event_engine::experimental::EventEngine::TaskHandle deadline_task_;
+  Mutex deadline_mu_;
+  Timestamp deadline_ ABSL_GUARDED_BY(deadline_mu_) = Timestamp::InfFuture();
+  grpc_event_engine::experimental::EventEngine::TaskHandle ABSL_GUARDED_BY(
+      deadline_mu_) deadline_task_;
+  // finished_ & completed_ represent the same state: is the call all done.
+  // finished_ is used for the promise returned by finished(), and can be
+  // awaited in a promise.
+  // completed_ is used for observing this state from outside of the activity
+  // (and hence outside the activity provided mutual exclusion).
   Latch<void> finished_;
```

Comment:
Shouldn't this now be replaced by `ExternallyObservableLatch`?

Reply:
yeah... for a minute it wasn't going to be possible (completed_ was in a base class); fixed